modules/io/test/perfImageLoadSave.cpp

```diff
@@ -0,0 +1,281 @@
+#include <catch.hpp>
+
+#include <thread>
+#include <visp3/core/vpIoTools.h>
+#include <visp3/io/vpImageIo.h>
+
+static std::string ipath = vpIoTools::getViSPImagesDataPath();
+static std::vector<std::string> paths {
+  ipath + "/Solvay/Solvay_conference_1927_Version2_640x440",
+  ipath + "/Solvay/Solvay_conference_1927_Version2_1024x705",
+  ipath + "/Solvay/Solvay_conference_1927_Version2_1280x881",
+  ipath + "/Solvay/Solvay_conference_1927_Version2_2126x1463",
+};
+static std::vector<std::string> names {
+  "Solvay (640x440)", "Solvay (1024x705)", "Solvay (1280x881)", "Solvay (2126x1463)"
+};
+static std::vector<vpImageIo::vpImageIoBackendType> backends {
+#if defined(VISP_HAVE_JPEG) && defined(VISP_HAVE_PNG)
+  vpImageIo::IO_LIB_BACKEND,
+#endif
+#if defined(VISP_HAVE_OPENCV)
+  vpImageIo::IO_OPENCV_BACKEND,
+#endif
+  vpImageIo::IO_SIMDLIB_BACKEND,
+  vpImageIo::IO_STB_IMAGE_BACKEND
+};
+static std::vector<std::string> backendNamesJpeg {
+  "libjpeg", "OpenCV", "simd", "stb"
+};
+static std::vector<std::string> backendNamesPng {
+  "libpng", "OpenCV", "simd", "stb"
+};
+static int nThreads = 0;
+
+TEST_CASE("Benchmark grayscale JPEG image loading", "[benchmark]") {
+  for (size_t i = 0; i < paths.size(); i++) {
+    SECTION(names[i]) {
+      for (size_t j = 0; j < backends.size(); j++) {
+        vpImage<vpRGBa> I;
+
+        BENCHMARK(backendNamesJpeg[j] + " backend") {
+          vpImageIo::read(I, paths[i] + ".jpg", backends[j]);
+          return I;
+        };
+      }
+    }
+  }
+}
+
+TEST_CASE("Benchmark RGBA JPEG image loading", "[benchmark]") {
+  for (size_t i = 0; i < paths.size(); i++) {
+    SECTION(names[i]) {
+      for (size_t j = 0; j < backends.size(); j++) {
+        vpImage<unsigned char> I;
+
+        BENCHMARK(backendNamesJpeg[j] + " backend") {
+          vpImageIo::read(I, paths[i] + ".jpg", backends[j]);
+          return I;
+        };
+      }
+    }
+  }
+}
+
+TEST_CASE("Benchmark grayscale PNG image loading", "[benchmark]") {
+  for (size_t i = 0; i < paths.size(); i++) {
+    SECTION(names[i]) {
+      for (size_t j = 0; j < backends.size(); j++) {
+        vpImage<vpRGBa> I;
+
+        BENCHMARK(backendNamesPng[j] + " backend") {
+          vpImageIo::read(I, paths[i] + ".png", backends[j]);
+          return I;
+        };
+      }
+    }
+  }
+}
+
+TEST_CASE("Benchmark RGBA PNG image loading", "[benchmark]") {
+  for (size_t i = 0; i < paths.size(); i++) {
+    SECTION(names[i]) {
+      for (size_t j = 0; j < backends.size(); j++) {
+        vpImage<unsigned char> I;
+
+        BENCHMARK(backendNamesPng[j] + " backend") {
+          vpImageIo::read(I, paths[i] + ".png", backends[j]);
+          return I;
+        };
+      }
+    }
+  }
+}
+
+std::string username, directory_filename_tmp;
+
+#if !defined(_WIN32) && (defined(__unix__) || defined(__unix) || (defined(__APPLE__) && defined(__MACH__))) // UNIX
+std::string tmp_dir = "/tmp/";
+#else
+std::string tmp_dir = "C:/Temp/";
```

Comment:
Could be updated since now there is `makeTempDirectory()` implemented for Windows.

Reply:
Resolved in commit 7c81377